### PR TITLE
Add cgroup memory dir check in gpcheckresgroupimpl.

### PR DESCRIPTION
**Pull request by kainwen**

### CODE REVIEW

File: gpMgmt/bin/gpcheckresgroupimpl [hunk 1]
```diff
@@ -49,6 +49,8 @@ class cgroup(object):
         self.validate_permission("cpuacct/gpdb/cpuacct.usage", "r")
         self.validate_permission("cpuacct/gpdb/cpuacct.stat", "r")
 
+        self.validate_permission("memory/", "rwx")
+
```
Comment (pivotal-ning-yu) [inline gpMgmt/bin/gpcheckresgroupimpl]: IIRC what we need under `cgroup/memory/` is two property files, we need to read the memory and swap limitation from them.

So we shouldn't expect to have `rwx` permission on `cgroup/memory/` as we don't want to list contents under this dir (`r`), create subdirs under it (`wx`) or chdir to it (`x`), we should only check whether we have `r` permission on the two property files.

Comment (kainwen) [inline gpMgmt/bin/gpcheckresgroupimpl]: It seems that we just need `memory` exists and we can pass it. So we just need `x`(for reading content in `memroy/xxxx`).

Comment (kainwen) [inline gpMgmt/bin/gpcheckresgroupimpl]: here is a test case:
```bash
Pivotals-MBP-2:a zlv$ chmod -x b
Pivotals-MBP-2:a zlv$ cat b/haha
cat: b/haha: Permission denied
Pivotals-MBP-2:a zlv$ chmod +x b
Pivotals-MBP-2:a zlv$ cat b/haha
abc
Pivotals-MBP-2:a zlv$ chmod -r b
Pivotals-MBP-2:a zlv$ cat b/haha
abc
```

Comment (pivotal-ning-yu) [inline gpMgmt/bin/gpcheckresgroupimpl]: When you removed `x` from b you also be removed `r` from b/haha, you can verify this as below:

```sh
$ mkdir dir
$ touch dir/file

# now we have 'r' perm on dir/file
$ test -r dir/file; echo $?
0

$ chmod -x dir
# now we lacks 'r' perm on dir/file
$ test -r dir/file; echo $?
1
```

**Comment by pivotal-ning-yu** [comment]
LGTM

**Pull request merged**